db/schema_applier.cc

```diff
@@ -957,12 +957,12 @@ void schema_applier::commit_tables_and_views() {
 
     for (auto& schema : tables.created) {
         auto& ks = db.find_keyspace(schema->ks_name());
-        db.add_column_family(ks, schema, ks.make_column_family_config(*schema, db), replica::database::is_new_cf::yes, _pending_token_metadata.local());
+        co_await db.add_column_family(ks, schema, ks.make_column_family_config(*schema, db), replica::database::is_new_cf::yes, _pending_token_metadata.local());
```

Comment:
This makes this method non-atomic, which might be a problem, considering it has "commit" in its name.

Reply:
Maybe we should have an async lock in schema_applier to enforce atomicity.
Keeping those functions synchronous makes them susceptible to reactor stalls.

Reply:
Ok, I just wanted to make sure we considered this aspect and this change is safe.

Reply:
schema_applier::commit() is async, because it's a multi-shard gather, but commit_on_shard() must not be, because we need to publish all schema and topology changes instantly on a given shard.  

The comment on schema_applier says:
```
It does atomic (per shard) application of a new schema.
```

You need to construct owned ranges in the earlier preparation phase.

\cc @nuivall